plugins/event/sqs/sqs.go

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package sqs
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/credentials"
+	"github.com/aws/aws-sdk-go-v2/service/sqs"
+	"github.com/hashicorp/vault/sdk/event"
+	"github.com/hashicorp/vault/version"
+)
+
+var _ event.EventSubscriptionPlugin = (*SQS)(nil)
+
+const pluginType = "sqs"
+
+func New() *SQS {
+	return &SQS{
+		clients:   map[string]*sqs.Client{},
+		queueUrls: map[string]string{},
+	}
+}
+
+type SQS struct {
+	clients    map[string]*sqs.Client
+	queueUrls  map[string]string
+	clientLock sync.RWMutex
+}
+
+func (s *SQS) Initialize(_ context.Context) error {
```

Comment:
Have you got use-cases in mind for Initialize on other plugins? Given the data model, it seems like Subscribe ends up taking most of what you'd normally consider part of initialisation (which is fine, I'm wondering if we should delete Initialize).

Reply:
You're right, I think we can drop `Initialize()`. I kept it mostly to match the interface of other other plugins, but it's not necessary.